smac_planner/src/a_star.cpp

```diff
@@ -315,42 +318,24 @@ void AStarAlgorithm::getNeighbors(const unsigned int & node, NodeVector & neighb
   // 100 100 100   where lower-middle '100' is visited with same cost by both bottom '50' nodes
   // Therefore, it is valuable to have some low-potential across the entire map
   // rather than a small inflation around the obstacles
-  const int node_i = static_cast<int>(node);
 
   switch (_neighborhood) {
     case Neighborhood::UNKNOWN:
       throw std::runtime_error("Unkown neighborhood type selected.");
-    case Neighborhood::VAN_NEUMANN:
-      getValidNodes(_van_neumann_neighborhood, node_i, neighbors);
+    case Neighborhood::VON_NEUMANN:
+      node->getNeighbors(_von_neumann_neighborhood, neighbors, _graph.get());
       break;
     case Neighborhood::MOORE:
-      getValidNodes(_moore_neighborhood, node_i, neighbors);
+      node->getNeighbors(_moore_neighborhood, neighbors, _graph.get());
       break;
     default:
       throw std::runtime_error("Invalid neighborhood type selected.");
   }
 }
 
-void AStarAlgorithm::getValidNodes(
-  const std::vector<int> & lookup_table,
-  const int & node_i,
-  NodeVector & neighbors)
-{
-  Node * node;
-  int index = 0;
-
-  for (unsigned int i = 0; i != lookup_table.size(); i++) {
-    index = node_i + lookup_table[i];
-    if (index > 0 && isNodeValid(index, node))
```

Comment:
Can you highlight specifically where you moved the initial validity check to?

Reply:
The validity check was moved to the `getNeighbors` method of the node class, [here](https://github.com/SteveMacenski/navigation2/pull/2/files#diff-39046cc286d73ec137d4639a25ac4e98R166-R172)